src/librustc/metadata/filesearch.rs

```diff
@@ -295,7 +294,7 @@ fn find_libdir(sysroot: &Path) -> String {
 
 #[cfg(windows)]
 fn find_libdir(_sysroot: &Path) -> String {
-    "bin".to_string()
+    "lib".to_string()
```

Comment:
I think with this change we could actually remove the `#[cfg]` from this function altogether perhaps?